5-app-infra/modules/env_base/main.tf

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  environment_code = element(split("", var.environment), 0)
+}
+
+resource "google_service_account" "compute_engine_service_account" {
+  project      = data.google_project.env_project.project_id
+  account_id   = "sa-compute-engine"
+  display_name = "Compute Engine service Account"
```

Comment:
Let's name it something like `Example App SA` to make users in console aware that this is not the default compute SA

Reply:
ok!